src/Http/Routing/src/Builder/EndpointRoutingApplicationBuilderExtensions.cs

```diff
@@ -118,9 +132,9 @@ private static void VerifyRoutingServicesAreRegistered(IApplicationBuilder app)
             }
         }
 
-        private static void VerifyEndpointRoutingMiddlewareIsRegistered(IApplicationBuilder app, out DefaultEndpointRouteBuilder endpointRouteBuilder)
+        private static void VerifyEndpointRoutingMiddlewareIsRegistered(IApplicationBuilder app, out IEndpointRouteBuilder endpointRouteBuilder)
         {
-            if (!app.Properties.TryGetValue(EndpointRouteBuilder, out var obj))
+            if (!app.Properties.TryGetValue(EndpointRouteBuilder, out var obj) && !app.Properties.TryGetValue(GlobalEndpointRouteBuilderKey, out obj))
```

Comment:
The above comment would allow you to remove this check as well.
```suggestion
            if (!app.Properties.TryGetValue(EndpointRouteBuilder, out var obj))
```